strings-csharp/SubstringInString/SubstringInString/SubstringSearchMethods.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Text.RegularExpressions;
+
+public static partial class SubstringSearchMethods
+{
+    public static List<int> FindAllIndexesWithSubstring(string input, string search)
+    {
+        List<int> indexes = new();
+
+        for (var i = 0; i <= input.Length - search.Length; i++)
+        {
+            if (input.Substring(i, search.Length) == search)
+            {
+                indexes.Add(i);
+            }
+        }
+
+        return indexes;
+    }
+
+    public static void PrintResult(List<int> result)
+    {
+        Console.WriteLine("Indexes where the substring is found: " + string.Join(", ", result));
+    }
+
+    public static List<int> FindAllIndexesWithSpan(string input, string search)
+    {
+        List<int> indexes = new();
+
+        var inputSpan = input.AsSpan();
+        var searchSpan = search.AsSpan();
+
+        for (int i = 0; i <= input.Length - search.Length; i++)
+        {
+            if (inputSpan.Slice(i, search.Length).SequenceEqual(searchSpan))
+            {
+                indexes.Add(i);
+            }
+        }
+
+        return indexes;
+    }
+
+    public static List<int> FindAllIndexesWithIndexOf(string input, string search)
```

Comment:
I would expect the performance to be slightly better here if we change the parameter types to `ReadOnlySpan<char>`.